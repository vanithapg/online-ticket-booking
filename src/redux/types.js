export const GET_MOVIES_REQUESTED = "GET_MOVIES_REQUESTED";
export const GET_MOVIES_SUCCESS = "GET_MOVIES_SUCCESS";
export const GET_MOVIES_FAILED = "GET_MOVIES_FAILED";

export const GET_MOVIEDETAILS_REQUESTED = "GET_MOVIEDETAILS_REQUESTED";
export const GET_MOVIEDETAILS_SUCCESS = "GET_MOVIEDETAILS_SUCCESS";
export const GET_MOVIEDETAILS_FAILED = "GET_MOVIEDETAILS_FAILED";
