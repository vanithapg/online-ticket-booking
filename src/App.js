import { Link, Route, BrowserRouter as Router } from "react-router-dom";
import React, { useState } from "react";

import "./App.css";
// import FoodItems from "./FoodItems";
// import Recipe from "./Recipe";
import Movie from "./Components/Movie";
import HomePage from "./Components/HomePage";
import Navbar from "./Components/Navbar";
import BookMovie from "./Components/BookMovie";
function App() {
  // const [foodItems, setFoodItems ] = useState([]);

  // useEffect(() => {
  //     fetch("http://localhost:6800/food", { method: "GET" })
  //     .then((response) => {
  //       return response.json();
  //     })
  //     .then((data) => {
  //         console.log(data);
  //       setFoodItems(data);
  //     });
  // }, [])

  return (
    <div className="App">
      <Navbar />
      <Router>
        <Route exact path="/" component={HomePage} />
        <Route path="/moviedetails/:id" component={Movie}></Route>
        <Route path="/bookmovie/:id" component={BookMovie}></Route>
      </Router>
    </div>
  );
}

export default App;
