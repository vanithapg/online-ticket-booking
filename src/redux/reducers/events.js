import * as type from "../types";

const initialState = {
  events: [],
  loading: false,
  error: null,
};

export default function events(state = initialState, action) {
  switch (action.type) {
    case type.GET_EVENTS_REQUESTED:
      return {
        ...state,
        loading: true,
      };
    case type.GET_EVENTS_SUCCESS:
      return {
        ...state,
        loading: false,
        events: action.events,
      };
    case type.GET_EVENTS_FAILED:
      return {
        ...state,
        loading: false,
        error: action.message,
      };
    default:
      return state;
  }
}
