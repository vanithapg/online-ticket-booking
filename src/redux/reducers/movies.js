import * as type from "../types";

const initialState = {
  movies: [],
  loading: false,
  error: null,
};

export default function movies(state = initialState, action) {
  switch (action.type) {
    case type.GET_MOVIES_REQUESTED:
      return {
        ...state,
        loading: true,
      };
    case type.GET_MOVIES_SUCCESS:
      return {
        ...state,
        loading: false,
        movies: action.movies,
      };
    case type.GET_MOVIES_FAILED:
      return {
        ...state,
        loading: false,
        error: action.message,
      };
    case type.GET_MOVIEDETAILS_REQUESTED:
      return {
        movies: state.movies.find((f) => f.id === action.value.id),
        loading: false,
        error: action.message,
      };
    case type.GET_MOVIEDETAILS_SUCCESS:
      return {
        ...state,
        loading: false,
        movies: action.movies,
      };
    case type.GET_MOVIEDETAILS_FAILED:
      return {
        ...state,
        loading: false,
        error: action.message,
      };
    default:
      return state;
  }
}
