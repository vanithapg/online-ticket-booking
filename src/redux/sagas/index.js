import { all } from "redux-saga/effects";
import recMovies from "./recommendedSagas";
import latest from "./latest";
import upcoming from "./upcoming";
import events from "./events";

export default function* rootSaga() {
  yield all([recMovies(), latest(), upcoming(), events()]);
}
