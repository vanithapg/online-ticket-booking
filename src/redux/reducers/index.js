import { combineReducers } from "redux";
import movies from "./movies";

const rootReducer = combineReducers({
  movies: movies,
});

export default rootReducer;
