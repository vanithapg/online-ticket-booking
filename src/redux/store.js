import { createStore, applyMiddleware, compose } from 'redux';
import createSagaMiddleware from 'redux-saga';
import rootReducer from './reducers';
import rootSaga from './sagas';

const sagaMiddleware = createSagaMiddleware();
const store = compose(
  applyMiddleware(sagaMiddleware),
  window.devToolsExtension && window.devToolsExtension(),
)(createStore)(rootReducer);

sagaMiddleware.run(rootSaga);

export default store;