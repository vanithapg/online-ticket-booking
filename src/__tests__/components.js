import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import { createStore } from "redux";
import reducers from "../redux/reducers";
import App from "../App";
import { useSelector, useDispatch } from "react-redux";

const store = createStore(reducers);
const component = renderer.create(
  <Provider store={store}>
    <App />
  </Provider>
);

describe("testing components", () => {
  test("no fooditems snapshot", () => {
    expect(component.toJSON()).toMatchSnapshot();
  });

  test("with food items snapshot", () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
