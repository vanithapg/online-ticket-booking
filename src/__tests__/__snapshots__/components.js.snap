// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing components no fooditems snapshot 1`] = `
<div
  className="App"
>
  <h1>
     Food Blog
  </h1>
  <hr />
  <p>
    No food recipies available!
  </p>
</div>
`;

exports[`testing components with food items snapshot 1`] = `
<div
  className="App"
>
  <h1>
     Food Blog
  </h1>
  <hr />
  <p>
    No food recipies available!
  </p>
</div>
`;
